reflect_map.go

```diff
@@ -310,7 +310,7 @@ func (encoder *sortKeysMapEncoder) Encode(ptr unsafe.Pointer, stream *Stream) {
 		encoder.elemEncoder.Encode(elem, subStream)
 		keyValues = append(keyValues, encodedKV{
```

Comment:
This line is the other large source of allocations for this code. Inside this function do we have access to the length of mapIter? If so, this slice could be correctly sized when it is instantiated.

Reply:
Initializing the `keyValues` slice to be size 8 saved another 25% in bytes allocated in my test. 8 will not always be the best value, but I think it's better than an initial size of 0 in most cases.

To get the actual length of the map, looks like reflect2 would need to expose [maplen](https://github.com/golang/go/blob/master/src/reflect/value.go#L2733).